### PR TITLE
spirv-val: label tests for VUID 04659

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-StandaloneSpirv-OpImageQuerySizeLod-04659

### CODE REVIEW

File: source/val/validate_image.cpp [hunk 1]
```diff
@@ -887,6 +887,7 @@ spv_result_t ValidateTypeSampledImage(ValidationState_t& _,
   // Vulkan uses the Sampled=1 case.
   if ((info.sampled != 0) && (info.sampled != 1)) {
     return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << _.VkErrorID(4659)
```
Comment (alan-baker) [inline source/val/validate_image.cpp:890]: This doesn't seem right. This code validates OpTypeSampledImage, not the opcodes listed in the VU. This VU is already checked in `ValidateImageQuerySizeLod` and `ValidateImageQueryLevels`.

File: test/val/val_image_test.cpp [hunk 2]
```diff
@@ -3893,6 +3893,8 @@ OpFunctionEnd
 
   CompileSuccessfully(body.c_str());
   ASSERT_EQ(SPV_ERROR_INVALID_DATA, ValidateInstructions(SPV_ENV_VULKAN_1_0));
+  EXPECT_THAT(getDiagnosticString(),
```
Comment (alan-baker) [inline test/val/val_image_test.cpp:3896]: These checks should stay.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: source/val/validate_image.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply source/val/validate_image.cpp:890]: the issue is without this the `OpImageQueryLod` usage fails (`ValidateImage.QueryLodVulkanNonSampledError`)

As you pointed [out here](https://github.com/KhronosGroup/SPIRV-Tools/pull/4103#discussion_r556545291), should there now be an explicit check in `ValidateImageQueryLod()`?

Comment (alan-baker) [reply source/val/validate_image.cpp:890]: What about changing this check slightly to cover part of VUID 04657 instead? My concern with using 04659 is that it mentions specific opcodes which may not be involved at all.

Comment (sfricke-samsung) [reply source/val/validate_image.cpp]: agree

**Review by alan-baker — APPROVED** [review]

**Pull request merged**